### PR TITLE
feat: add all default assets to aspects-superset image to avoid load it via larger file

### PR DESCRIPTION
**Pull request by Henrrypg**
# Description

This PR intends to modify `aspects-superset` image to contains all default `openedx-assets` and load it from there, and still load custom assets with the patch over `superset-extra-assets`. This changes will keep a small `assets.yaml` file and ConfigMap for K8S,

## How to test it

You should run an openedx environment from cero without any problem in the init.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Henrrypg!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by Henrrypg** [comment]
@Ian2012 Can you please take a look at this?

File: tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py [hunk 1]
```diff
@@ -61,26 +62,45 @@ def main():
...
+                # Process the asset directly
+                if FILE_NAME_ATTRIBUTE in asset:
+                    file_name = asset.pop(FILE_NAME_ATTRIBUTE)
+
+                    # Find the right folder to create the asset in
+                    for asset_name, folder in ASSET_FOLDER_MAPPING.items():
+                        if asset_name in asset:
+                            write_asset_to_file(asset, asset_name, folder, file_name, roles)
+                            break
+
```
Comment (Ian2012) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py:86]: Can you use the circuit breaker pattern?

```python
if not condition:
    break / continue
```

File: tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py [hunk 2]
```diff
@@ -61,26 +62,45 @@ def main():
...
     with open(ASSETS_FILE_PATH, "r") as file:
         extra_assets = yaml.safe_load(file)
 
-        if not extra_assets:
-            print("No extra assets to create")
-            return
+        if extra_assets:
+            # For each asset, create a file in the right folder
+            for asset in extra_assets:
+                if FILE_NAME_ATTRIBUTE not in asset:
```
Comment (Ian2012) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py:93]: Same comment as above. You can restore the condition and use a break statement

**Review by Ian2012 — CHANGES_REQUESTED** [review]
The sql files can be removed from the docker image and can live only in the plugin templates.

File: tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py (hunk 2 above)
Comment (Henrrypg) [reply tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py:93]: `break` or `continue` statement do not work for non loop statements. I have being searching and not appear have a way to skip only the `with` statement. Any ideas?

Comment (Ian2012) [reply tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py:94]: Leave it as it's

File: tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py [hunk 3]
```diff
@@ -61,26 +62,46 @@ def main():
...
+                        write_asset_to_file(asset, asset_name, folder, file_name, roles)
+                        break
+
     with open(ASSETS_FILE_PATH, "r") as file:
         extra_assets = yaml.safe_load(file)
 
-        if not extra_assets:
-            print("No extra assets to create")
-            return
+        if extra_assets:
```
Comment (Ian2012) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py:91]: Now that the extra_assets is empty by default, did this worked if empty?

Comment (Henrrypg) [reply tutoraspects/templates/aspects/apps/superset/pythonpath/create_assets.py:91]: Yes, it work correctly with `extra_assets` empty

**Review by Ian2012 — APPROVED** [review]
@bmtcril do you want to take a look?

@pomegranited you can rebase over this changes on your PR

File: tutoraspects/templates/aspects/apps/superset/pythonpath/test.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import yaml
```
Comment (bmtcril) [inline tutoraspects/templates/aspects/apps/superset/pythonpath/test.py:1]: Is this file still needed?

**Review by bmtcril — COMMENTED** [review]
Just the one question. We should also update documentation to indicate that any time an asset or translation change is made we will need to rebuild the image before re-running init to test now.

**Comment by Ian2012** [comment]
@bmtcril why translations?

Those are still loaded from a config file

**Comment by bmtcril** [comment]
@Ian2012 true, but I think the point about assets still stands

**Comment by Henrrypg** [comment]
@bmtcril Somehing like that? https://github.com/openedx/tutor-contrib-aspects/pull/500/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168f

File: README.rst [hunk 5]
```diff
@@ -153,6 +153,7 @@ To contribute assets to Aspects:
 #. The script will also warn about missing `_roles` in dashboards. Superset does not export
    these, so you will need to manually add this key with the roles that are necessary to
    view the dashboard. See the existing dashboards for how this is done.
+#. Re-build your openedx-assets image with `tutor images build openedx-assets --no-cache`
```
Comment (Ian2012) [inline README.rst:156]:
```suggestion
#. Re-build your openedx-assets image with `tutor images build superset --no-cache`
```

**Review by Ian2012 — CHANGES_REQUESTED** [review]

File: README.rst (hunk 5 above)
Comment (Henrrypg) [reply README.rst]: My bad, but is `aspects-superset`, right?

Comment (Ian2012) [reply README.rst]: yes

**Review by Ian2012 — APPROVED** [review]

**Comment by Henrrypg** [comment]
One test is failing, it seems that aspects-superset image is not being built on workflow, maybe we need to add this step? @Ian2012  @bmtcril

**Comment by Ian2012** [comment]
@Henrrypg the error is caused by a flaky migration in Superset. The workflow is re run

**Comment by Ian2012** [comment]
@Henrrypg the error is caused by a flaky migration in Superset. The workflow is re run

**Review by bmtcril — APPROVED** [review]
Bah sorry, I merged another PR before this one. You'll need to rebase and probably move the new files over to the new location. This looks ready to merge once that's done, though.

**Comment by openedx-webhooks** [comment]
@Henrrypg 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/500&entry.1484655318=https://github.com/Henrrypg&entry.752974735=2023-10-26+21:32&entry.1917517419=2023-11-01+17:33&entry.2133058324=Yes) so we can improve your experience in the future.
